Core/clim-basic/extended-streams/text-formatting.lisp

```diff
@@ -22,7 +22,7 @@
   (ignore-errors ; destructuring-bind may error; that yields invalid spec
     (destructuring-bind (&key left top right bottom) margins
       (flet ((margin-spec-p (m)
-               (and (member (first m) '(:absolute :relative))
+               (and (member (first m) '(:relative :absolute))
```

Comment:
Can you add a `destructuring-bind` to check the shape of `m` and document the meaning of its parts?

Reply:
given it looks like this:
```
(defun valid-margin-spec-p (margins)
  (ignore-errors ; destructuring-bind may error; that yields invalid spec
   (destructuring-bind (&key left top right bottom) margins
     (flet ((margin-spec-p (margin)
              (destructuring-bind (anchor value) margin
                (and (member anchor '(:relative :absolute))
                     (realp value)))))
       (every #'margin-spec-p (list left top right bottom))))))
```

how would you like me to document it?

(notice, that I've changed (not (null value)) to (realp value))

Reply:
`destructuring-bind (anchor value) margin` is exactly what I meant by documenting the shape using `destructuring-bind`.